### PR TITLE
reproduce #560

### PR DESCRIPTION
**Pull request by TENX-S**

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job!

**Comment by fzyzcjy** [comment]
Seems like an enum problem. 

/cc @Desdaemon who implemented this great feature. What do you think?

File: frb_example/pure_dart/rust/src/bridge_generated.rs [hunk 1]
```diff
@@ -1085,6 +1112,11 @@ pub extern "C" fn new_box_u8_0(value: u8) -> *mut u8 {
     support::new_leak_box_ptr(value)
 }
 
+#[no_mangle]
+pub extern "C" fn new_box_weekdays_0() -> *mut i32 {
+    support::new_leak_box_ptr(i32::new_with_null_ptr())
```
Comment (Desdaemon) [inline frb_example/pure_dart/rust/src/bridge_generated.rs:1117]: This should be `*mut i32`

**Review by Desdaemon — COMMENTED** [review]
I probably missed a pointer check in the gen-code for Box, I'll take a look at this soon.

**Pull request merged**